wordpress-mobile/WordPress-Android
# Add stats email deeplink handling

Fixes #14449 

This PR adds handling of email links like `public-api.wordpress.com/mbar/...redirect_to=wordpress.com/stats/day/example.com`. As addition I've pulled the logic for simple stats links like `wordpress.com/stats/week/example.com` to be handled by the same `StatsLinkHandler`. One more improvement is that the deeplinks now handle the stats timeframe as well as the site so if there is a link to `day`, it opens the day tab. Let me know if you need links to test this feature out. 

To test:
- Make sure you're logged in with an account that has the `your_site.com` site 
- Open a deeplink like `public-api.wordpress.com/mbar/...redirect_to=wordpress.com/stats/day/your_site.com`
- Notice it's opened in the app
- Notice it opens the Stats screen with the "Day" tab selected for `your_site.com`
- Test this with replacing `day` in the link with `week`, `year`, `insights` for all the screens

To test:
- Make sure you're not logged in with an account that has the `another_site.com` site 
- Open a deeplink like `public-api.wordpress.com/mbar/...redirect_to=wordpress.com/stats/day/another_site.com`
- Notice it's opened in the app
- Notice it opens the Stats screen for the currently selected site

To test:
- Make sure you're logged in with an account that has the `your_site.com` site 
- Open a deeplink like `wordpress.com/stats/day/your_site.com`
- Notice it's opened in the app
- Notice it opens the Stats screen with the "Day" tab selected for `your_site.com`

To test:
- Open a deeplink like `public-api.wordpress.com/bar/...redirect_to=wordpress.com/stats/day/your_site.com`
- Notice it's opened in the browser

## Regression Notes
1. Potential unintended areas of impact
- none 🤞 

PR submission checklist:
